File: 프로그래머스/lv1/12977. 소수 만들기/소수 만들기.kt
class Solution {
    fun solution(nums: IntArray): Int {
        var answer = 0

        for(i in 0 until nums.size-2) {
            for(j in i+1 until nums.size-1) {
                for(k in j+1 until nums.size) {
                    if(minorityStatus(nums[i] + nums[j] + nums[k])) {
                        answer++
                    }
                }
            }
        }
        return answer
    }

    fun minorityStatus(num: Int): Boolean {
        for(i in 2 until num) {
            if(num % i == 0) {
                return false
            }
        }
        return true
    }
}

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.swift
func solution(_ nums:[Int]) -> Int {
    var answer = 0
    
    for i in 0...nums.count-3 {
        for j in i+1...nums.count-2 {
            for k in j+1...nums.count-1 {
                answer += checkPrimeNumber(nums[i]+nums[j]+nums[k])
            }
        }
    }
    
    return answer
}

func checkPrimeNumber(_ num: Int) -> Int {
    for i in 2...num/2 {
        if num % i == 0 {
            return 0
        }
    }
    
    return 1
}